ckanext/emailasusername/plugin.py

```diff
@@ -88,3 +93,25 @@ def user_emails_match(key, data, errors, context):
         )
     else:
         data[('email',)] = email1
+
+
+@classmethod
+def search_by_username_and_email(cls, querystr, sqlalchemy_query=None, user_name=None):
```

Comment:
This method is defined outside of any class. Decorator `@classmethod` doesn't fit here.
After thinking about it for a while I think it's too hacky way to overwrite `User.search` by this substitution:
```
User.search = search_by_username_and_email
```
Could you explain a bit why you took this approach? Are there some places in CKAN which do things similarly?